src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1187,17 +1194,16 @@ QVector<qreal> TorrentImpl::filesProgress() const
     if (m_completedFiles.count(true) == filesCount())
         return QVector<qreal>(filesCount(), 1);
 
-    std::vector<int64_t> fp;
-    m_nativeHandle.file_progress(fp, lt::torrent_handle::piece_granularity);
+    if (m_filesProgress.isEmpty())
+        return {};
 
     const int count = filesCount();
```

Comment:
I saw that `TorrentContentModel::updateFilesProgress()` has this:
```c++
Q_ASSERT(m_filesIndex.size() == filesProgress.size());
// XXX: Why is this necessary?
if (Q_UNLIKELY(m_filesIndex.size() != filesProgress.size()))
    return;
```

Does it make sense to put it here too?

Reply:
>Does it make sense to put it here too?

Do you mean add `Q_ASSERT(!m_filesProgress.isEmpty())`?
We shouldn't assert on it, since it is one of regular cases. It is empty when torrent still has no metadata.

Reply:
No. Let me adjust the code to the current names:
```c++
Q_ASSERT(count) == m_filesProgress.size());
// XXX: Why is this necessary?
if (Q_UNLIKELY(count != m_filesProgress.size()))
    return;
```

Technically the for-loop uses an iterator based on `count` but iterates over `m_filesProgress` too.

Reply:
> Technically the for-loop uses an iterator based on `count` but iterates over `m_filesProgress` too.

OK. Reformatted it and added some assertion.